packages/ember-data/lib/adapters/json-api-adapter.js

```diff
@@ -0,0 +1,84 @@
+/**
+  @module ember-data
+*/
+
+import RESTAdapter from "ember-data/adapters/rest-adapter";
+
+/**
+  @class JSONAPIAdapter
+  @constructor
+  @namespace DS
+  @extends DS.RESTAdapter
+*/
+export default RESTAdapter.extend({
+  defaultSerializer: '-json-api',
+
+  /**
+    @method ajaxOptions
+    @private
+    @param {String} url
+    @param {String} type The request type GET, POST, PUT, DELETE etc.
+    @param {Object} options
+    @return {Object}
+  */
+  ajaxOptions: function(url, type, options) {
+    let hash = this._super(...arguments);
+
+    if (hash.contentType) {
+      hash.contentType = 'application/vnd.api+json; charset=utf-8';
```

Comment:
I don't think parameters, like `charset=utf-8` are allowed by the spec right now. See [content negotiation](http://jsonapi.org/format/#content-negotiation).